### PR TITLE
refactor(source-zendesk-support): migrate to CDK v3

### PR DESCRIPTION
**Pull request by artem1205**
## What

migrate to CDK v3; 
Following https://github.com/airbytehq/airbyte/pull/39811

## How

migrate:
- `def backoff_time` -> `def get_backoff_strategy`
- `def should_retry` -> `def get_error_handler`

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py [hunk 1]
```diff
@@ -45,37 +48,36 @@ def __init__(self, **kwargs):
...
+                return retry_after
+
+            # the header X-Rate-Limit returns the amount of requests per minute
+            rate_limit = float(response_or_exception.headers.get("X-Rate-Limit", 0))
+            if rate_limit and rate_limit > 0:
+                return 60.0 / rate_limit
+        return None
+
+
+class ZendeskSupportHttpStatusErrorHandler(HttpStatusErrorHandler):
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:67]: Why do we need this empty class?

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py [hunk 2]
```diff
@@ -121,20 +123,20 @@ def parse_response(self, response: requests.Response, stream_state: Mapping[str,
...
+                failure_type=FailureType.config_error,
+                error_message="Forbidden. Please ensure the authenticated user has access to this stream. If the issue persists, contact Zendesk support.",
+            ),
+            404: ErrorResolution(
+                response_action=ResponseAction.IGNORE,
+                failure_type=FailureType.config_error,
+                error_message="Not found. Please ensure the authenticated user has access to this stream. If the issue persists, contact Zendesk support.",
+            ),
+        }
+        return ZendeskSupportHttpStatusErrorHandler(logger=self.logger, max_retries=10, error_mapping=error_mapping)
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:139]: Can we instantiate from `HttpStatusErrorHandler` directly, instead?

**Review by bazarnov — APPROVED** [review]
Minor comments, otherwise looks good!

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py (hunk 2 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:135]: agreed, ref

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py (hunk 1 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py]: removed

**Pull request merged**